packages/eas-cli/src/update/configure.ts

```diff
@@ -0,0 +1,291 @@
+import { modifyConfigAsync } from '@expo/config';
+import { ExpoConfig } from '@expo/config-types';
+import { Platform, Workflow } from '@expo/eas-build-job';
+import chalk from 'chalk';
+
+import { getEASUpdateURL } from '../api';
+import { ExpoGraphqlClient } from '../commandUtils/context/contextUtils/createGraphqlClient';
+import { AppPlatform } from '../graphql/generated';
+import Log, { learnMore } from '../log';
+import { RequestedPlatform, appPlatformDisplayNames } from '../platform';
+import {
+  installExpoUpdatesAsync,
+  isExpoUpdatesInstalledOrAvailable,
+} from '../project/projectUtils';
+import { resolveWorkflowPerPlatformAsync } from '../project/workflow';
+import { syncUpdatesConfigurationAsync as syncAndroidUpdatesConfigurationAsync } from './android/UpdatesModule';
+import { syncUpdatesConfigurationAsync as syncIosUpdatesConfigurationAsync } from './ios/UpdatesModule';
+
+export const DEFAULT_MANAGED_RUNTIME_VERSION = { policy: 'sdkVersion' } as const;
+export const DEFAULT_BARE_RUNTIME_VERSION = '1.0.0' as const;
+
+function getDefaultRuntimeVersion(workflow: Workflow): NonNullable<ExpoConfig['runtimeVersion']> {
+  return workflow === Workflow.GENERIC
+    ? DEFAULT_BARE_RUNTIME_VERSION
+    : DEFAULT_MANAGED_RUNTIME_VERSION;
+}
+
+function isRuntimeEqual(
+  runtimeVersionA: NonNullable<ExpoConfig['runtimeVersion']>,
+  runtimeVersionB: NonNullable<ExpoConfig['runtimeVersion']>
+): boolean {
+  if (typeof runtimeVersionA === 'string' && typeof runtimeVersionB === 'string') {
+    return runtimeVersionA === runtimeVersionB;
+  } else if (typeof runtimeVersionA === 'object' && typeof runtimeVersionB === 'object') {
+    return runtimeVersionA.policy === runtimeVersionB.policy;
+  } else {
+    return false;
+  }
+}
+
+/**
+ * Make sure the `app.json` is configured to use EAS Updates.
+ * This does a couple of things:
+ *   - Ensure update URL is set to the project EAS endpoint
+ *   - Ensure runtimeVersion is defined for both or individual platforms
+ *   - Output the changes made, or the changes required to make manually
+ */
+export async function ensureEASUpdatesIsConfiguredInExpoConfigAsync({
+  exp,
+  projectId,
+  projectDir,
+  platform,
+  workflows,
+}: {
+  exp: ExpoConfig;
+  projectId: string;
+  projectDir: string;
+  platform: RequestedPlatform;
+  workflows: Record<Platform, Workflow>;
+}): Promise<{ projectChanged: boolean; exp: ExpoConfig }> {
+  const modifyConfig: Partial<ExpoConfig> = {};
+
+  if (exp.updates?.url !== getEASUpdateURL(projectId)) {
+    modifyConfig.updates = { url: getEASUpdateURL(projectId) };
+  }
+
+  let androidRuntimeVersion = exp.android?.runtimeVersion ?? exp.runtimeVersion;
```

Comment:
Not super important but we had a lot of this kinda logic in `expo export` and `expo publish` where everything was duplicated twice. Prefer loops even though there are only two values.

Reply:
One thing we can do here is changing the order of changes, much like how AST sanitization would work.

Instead of checking ahead of time if `runtimeVersions` are similar, we can do that after.
- Loop per platform first
  - Make required changes
- If requested platform === all && runtimeVersion is identical for all platforms: move it one level up

Does that sound better?